src/java.desktop/macosx/classes/sun/lwawt/macosx/CPlatformWindow.java

```diff
@@ -334,6 +334,10 @@ public void initialize(Window _target, LWWindowPeer _peer, PlatformWindow _owner
 
         responder = createPlatformResponder();
         contentView.initialize(peer, responder);
+        boolean isTranslucent = peer != null && peer.isTranslucent();
+        if (!isTranslucent) {
+            contentView.setWindowLayerOpaque(true);
+        }
```

Comment:
I think windowLayer should be always in sync with NSWindowOpaque state. And both should be changed together via setOpaque() method.

The change above will call the "setWindowLayerOpaque" twice:
 - LWWindowPeer()->platformWindow.initialize()->contentView.setWindowLayerOpaque(true)
 - LWWindowPeer()->initializeImpl()->setOpaque()->contentView.setWindowLayerOpaque()